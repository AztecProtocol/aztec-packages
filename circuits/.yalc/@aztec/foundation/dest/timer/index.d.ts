export { TimeoutTask } from './timeout.js';
export { Timer } from './timer.js';
//# sourceMappingURL=index.d.ts.map