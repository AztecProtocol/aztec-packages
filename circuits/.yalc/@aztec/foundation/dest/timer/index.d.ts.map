{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../src/timer/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,WAAW,EAAE,MAAM,cAAc,CAAC;AAC3C,OAAO,EAAE,KAAK,EAAE,MAAM,YAAY,CAAC"}