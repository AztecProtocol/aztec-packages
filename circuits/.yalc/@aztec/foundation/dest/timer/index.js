export { TimeoutTask } from './timeout.js';
export { Timer } from './timer.js';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi9zcmMvdGltZXIvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxFQUFFLFdBQVcsRUFBRSxNQUFNLGNBQWMsQ0FBQztBQUMzQyxPQUFPLEVBQUUsS0FBSyxFQUFFLE1BQU0sWUFBWSxDQUFDIn0=