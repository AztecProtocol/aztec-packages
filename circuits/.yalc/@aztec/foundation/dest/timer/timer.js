export class Timer {
    constructor() {
        this.start = new Date().getTime();
    }
    ms() {
        return new Date().getTime() - this.start;
    }
    s() {
        return (new Date().getTime() - this.start) / 1000;
    }
}
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidGltZXIuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi9zcmMvdGltZXIvdGltZXIudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsTUFBTSxPQUFPLEtBQUs7SUFHaEI7UUFDRSxJQUFJLENBQUMsS0FBSyxHQUFHLElBQUksSUFBSSxFQUFFLENBQUMsT0FBTyxFQUFFLENBQUM7SUFDcEMsQ0FBQztJQUVNLEVBQUU7UUFDUCxPQUFPLElBQUksSUFBSSxFQUFFLENBQUMsT0FBTyxFQUFFLEdBQUcsSUFBSSxDQUFDLEtBQUssQ0FBQztJQUMzQyxDQUFDO0lBRU0sQ0FBQztRQUNOLE9BQU8sQ0FBQyxJQUFJLElBQUksRUFBRSxDQUFDLE9BQU8sRUFBRSxHQUFHLElBQUksQ0FBQyxLQUFLLENBQUMsR0FBRyxJQUFJLENBQUM7SUFDcEQsQ0FBQztDQUNGIn0=