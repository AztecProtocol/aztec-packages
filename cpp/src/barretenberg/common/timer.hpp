#pragma once
#include <cstdio>
#include <string>
#include <sys/resource.h>
#include <sys/time.h>
#include <ctime>

/**
 * @brief Get the execution between a block of code.
 *
 */
class Timer {
  private:
    struct timespec _startTime;
    struct timespec _endTime;

    static constexpr int64_t NanosecondsPerSecond = 1000LL * 1000 * 1000;

    /**
     * @brief Manually sets the start time.
     */
    void start() { clock_gettime(CLOCK_REALTIME, &_startTime); }

    /**
     * @brief Manually sets the end time.
     */
    void end() { clock_gettime(CLOCK_REALTIME, &_endTime); }

  public:
    /**
     * @brief Initialize a Timer with the current time.
     *
     */
    Timer()
        : _endTime({})
    {
        start();
    }

    /**
     * @brief Return the number of nanoseconds elapsed since the start of the timer.
     */
    [[nodiscard]] int64_t nanoseconds() const
    {
        struct timespec end;
        if (_endTime.tv_nsec == 0 && _endTime.tv_sec == 0) {
            clock_gettime(CLOCK_REALTIME, &end);
        } else {
            end = _endTime;
        }

        int64_t nanos = (end.tv_sec - _startTime.tv_sec) * NanosecondsPerSecond;
        nanos += (end.tv_nsec - _startTime.tv_nsec);

        return nanos;
    }

    /**
     * @brief Return the number of seconds elapsed since the start of the timer.
     */
    [[nodiscard]] double seconds() const
    {
        int64_t nanos = nanoseconds();
        double secs = static_cast<double>(nanos) / NanosecondsPerSecond;
        return secs;
    }

    /**
     * @brief Return the number of seconds elapsed since the start of the timer as a string.
     */
    [[nodiscard]] std::string toString() const
    {
        double secs = seconds();
        return std::to_string(secs);
    }
};