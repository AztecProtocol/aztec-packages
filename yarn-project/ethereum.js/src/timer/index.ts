export class Timer {
  private start: number;

  constructor() {
    this.start = new Date().getTime();
  }

  public ms() {
    return new Date().getTime() - this.start;
  }

  public s() {
    return (new Date().getTime() - this.start) / 1000;
  }
}
