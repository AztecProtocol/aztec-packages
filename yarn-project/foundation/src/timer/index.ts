export { TimeoutTask } from './timeout.js';
export { Timer } from './timer.js';
