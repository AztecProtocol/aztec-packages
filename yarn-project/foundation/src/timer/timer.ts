/**
 * Timer class to measure time intervals in milliseconds and seconds.
 * Upon instantiation, it stores the current timestamp as the starting point.
 * The 'ms()' method returns the elapsed time in milliseconds,
 * while the 's()' method returns the elapsed time in seconds.
 *
 * @example
 * const timer = new Timer();
 * setTimeout(() =\> \{
 *   console.log(`Elapsed time: ${timer.ms()} ms`);
 * \}, 1000);
 */
export class Timer {
  private start: number;

  constructor() {
    this.start = new Date().getTime();
  }

  /**
   * Returns the elapsed time in milliseconds since the Timer instance was created.
   * Provides a simple and convenient way to measure the time duration between two events
   * or monitor performance of specific code sections.
   *
   * @returns The elapsed time in milliseconds.
   */
  public ms() {
    return new Date().getTime() - this.start;
  }

  /**
   * Returns the time elapsed since the Timer instance was created, in seconds.
   * The value is calculated by subtracting the initial start time from the current time
   * and dividing the result by 1000 to convert milliseconds to seconds.
   *
   * @returns The elapsed time in seconds.
   */
  public s() {
    return (new Date().getTime() - this.start) / 1000;
  }
}
