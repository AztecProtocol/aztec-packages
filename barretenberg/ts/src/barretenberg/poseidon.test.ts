import { BarretenbergSync } from './index.js';
import { Timer } from '../benchmark/timer.js';
import { Fr } from '../types/index.js';

describe('poseidon sync', () => {
  let api: BarretenbergSync;

  beforeAll(async () => {
    api = await BarretenbergSync.new();
  });

  it('poseidonHash', () => {
    const result = api.poseidonHash([new Fr(4n), new Fr(8n)]);
    expect(result).toMatchSnapshot();
  });

  it('poseidonHash perf test', () => {
    const loops = 1000;
    const fields = Array.from({ length: loops * 2 }).map(() => Fr.random());
    const t = new Timer();
    for (let i = 0; i < loops; ++i) {
      api.poseidonHash([fields[i * 2], fields[i * 2 + 1]]);
    }
    const us = t.us() / loops;
    console.log(`Executed ${loops} hashes at an average ${us}us / hash`);
  });

  it('poseidonHashes perf test', () => {
    const loops = 10;
    const numHashesPerLoop = 1024;
    const fields = Array.from({ length: numHashesPerLoop * 2 }).map(() => Fr.random());
    const t = new Timer();
    for (let i = 0; i < loops; ++i) {
      api.poseidonHashes(fields);
    }
    const us = t.us() / (numHashesPerLoop * loops);
    console.log(`Executed ${numHashesPerLoop * loops} hashes at an average ${us}us / hash`);
  });
});
