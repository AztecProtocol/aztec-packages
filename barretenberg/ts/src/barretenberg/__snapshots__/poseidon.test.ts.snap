// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poseidon sync poseidonHash 1`] = `
Fr {
  "value": Uint8Array [
    43,
    202,
    235,
    109,
    88,
    187,
    56,
    186,
    247,
    83,
    213,
    140,
    60,
    150,
    97,
    143,
    234,
    130,
    22,
    51,
    69,
    41,
    94,
    180,
    14,
    136,
    52,
    78,
    235,
    12,
    226,
    161,
  ],
}
`;
